### PR TITLE
Add procedure in consultation update

### LINKED ISSUE #3195: Adding Procedure to Consultation
In the consultation form (https://care.coronasafe.in/facility/015ad911-39b7-4104-b304-bf8178c33830/patient/aefb147a-eb09-4dd0-8390-e2cd5a05f326/consultation/c5e82e5f-2bab-4cab-9239-307770792bce/update), there needs to be a section for Procedures.

Suggestive design:
Screenshot 2022-07-22 at 5 15 49 PM

Content:
1. Repetitive: Toggle button for Yes/No
2. If repetitive, field for "Frequency" with dropdown  [ 15 min, 30 min, 1 hr, 6 hrs, 12 hrs, 24 hrs ]
    If not repetitive "Time and Date" to capture the time and date of the procedure
3. Procedure: Text field to accommodate 100 characters to describe the procedure
4. Note: Text field to accommodate 100 characters to add a note

**NOTE:** Earlier we had a simple text field in the consultation form for "Operations". It was later on taken off by https://github.com/coronasafe/care_fe/pull/2900. The same operation field may be used here for the Procedure

This section is to come after the Prescription/PRN Prescription Section

### PR DESCRIPTION
**Pull request by arpancodes**
fixes #3195
Added procedure components for consultation update.

image

---

image

### CODE REVIEW

**Comment by nihal467** [comment]
test approved

**Review by Ashesh3 — APPROVED** [review]

**Review by gigincg — CHANGES_REQUESTED** [review]
@arpancodes Using different code-formatting makes it almost impossible to review. Can you fix this?

**Comment by arpancodes** [comment]
> @arpancodes Using different code-formatting makes it almost impossible to review. Can you fix this?

The formatted files were actually unrelated to the PR, I guess they are old ones that's why prettier just changed every bit of those files.
Removed from the PR.

File: src/Components/Facility/ConsultationForm.tsx [hunk 1]
```diff
@@ -545,9 +600,19 @@ export const ConsultationForm = (props: any) => {
...
+  const handleDateChange = (
+    date: MaterialUiPickersDate,
+    key: string,
+    nested?: string[]
+  ) => {
+    let newValue: any = { [key]: date };
+    if (nested?.length) {
+      nested?.forEach((i: string) => {
+        newValue = { [i]: newValue };
+      });
```
Comment (khavinshankar) [inline src/Components/Facility/ConsultationForm.tsx:687]: just need some clarification here,
```js
// if we have something like this
handleDateChange(new Date(), 'now', ['procedure', 'extra']);

// then new value becomes
// initial
newValue = { now: new Date() }
// procedure
newValue = {procedure: { now: new Date() }}
// extra
newValue = {extra: {procedure: { now: new Date() }}}
```

So, if my understanding is correct, why are we nesting them?

File: src/Components/Facility/ConsultationForm.tsx [hunk 2]
```diff
@@ -492,6 +530,23 @@ export const ConsultationForm = (props: any) => {
         form: { ...state.form, [e.target.name]: e.target.value },
       });
   };
+  const handleProcedureChange: ChangeEventHandler<HTMLInputElement> = (
+    e: any
+  ) => {
+    console.log(e.target.value);
```
Comment (khavinshankar) [inline src/Components/Facility/ConsultationForm.tsx]:
```suggestion
```

Comment (khavinshankar) [reply src/Components/Facility/ConsultationForm.tsx]: @arpancodes just need some clarity

File: src/Components/Facility/ConsultationForm.tsx (hunk 1 above)
Comment (arpancodes) [reply src/Components/Facility/ConsultationForm.tsx:687]: Hey @khavinshankar, it was just an approach to handle the date changes for the nested elements as well... when they are not at the root level of the form object but inside another object.
Nesting just checks if that is the case and traces backwards and creates an updated object from that level to the root.

**Comment by aparnacoronasafe** [comment]
@arpancodes Can we close this ASAP?

**Comment by arpancodes** [comment]
@khavinshankar is everything fine?

**Comment by aparnacoronasafe** [comment]
@arpancodes I see changes requested. Please look into this and try closing it this week

**Comment by nihal467** [comment]
@aparnacoronasafe check the merge conflict, and merge latest development branch and check cypress is passing or not

**Comment by mathew-alex** [comment]
:wave: Hi, @arpancodes,
Conflicts have been detected against the base branch. Please rebase your branch against the base branch.

***

This message is automatically generated by [prince-chrismc/label-merge-conflicts-action](https://github.com/prince-chrismc/label-merge-conflicts-action) so don't hesitate to report issues/improvements there.

**Comment by nihal467** [comment]
@khavinshankar 
image

- [ ] currently, when you hover over yes/no, and select yes/no , until you don't hover out of yes/no, we can't identity which one is select. So, when we select any button, make the selected one green
- [ ] in the green background, make the text colour as white by replacing the existing grey colour
image

- [ ] even though we haven't selected the calendar icon, just after selecting no in repetition, below the icon it is mention invalid date ( occured when we try to update existing consultation form )

**Comment by nihal467** [comment]
@khavinshankar 

there are few issues need to take up for discussion in the EOD call

- [ ] the current system and proposed system designs are different

image

- [ ] Procedure and notes are limited to 100 character in the issue, so, allotting full individual rows for each of them, feel like waste of space. Below screenshot will give a example of space utilized by 100 character.
image
- [ ] since there is only yes/no , in the procedure section, anyone value by default kept pre-select in repetition and frequency field, without users input. For example, without filling any field in procedure section and submit the consultation field, the values in the below screenshot are by default saved by the form

image

**Comment by khavinshankar** [comment]
@nihal467 yes, it was done by @arpancodes long back, i'm not sure even if this is relevant anymore, we already investigation, so not sure if we need procedure

**Comment by nihal467** [comment]
@khavinshankar as per EOD, it is been redesign similar to investigation in the consultation page

**Comment by khavinshankar** [comment]
this feature has already been added

**Pull request closed without merging**